### PR TITLE
prometheus-apcupsd-exporter: unstable-2019-03-14 -> 0.2.0

### PR DESCRIPTION
**Pull request by mdlayher**
Signed-off-by: Matt Layher <mdlayher@gmail.com>

###### Motivation for this change

- I've done some work to bring apcupsd_exporter up to modern Go/Prometheus standards and the existing version is probably pretty broken
- Switch to buildGoModule and tagged release
- Add myself as a maintainer

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Binary builds and runs successfully.

```
[matt@servnerr-3:~/src/nixpkgs]$ nix-build -A prometheus-apcupsd-exporter
these derivations will be built:                                                                 
  /nix/store/vxfj1zydbsmh5j7sg5wn7av9l42drjaq-apcupsd-exporter-0.2.0.drv
building '/nix/store/vxfj1zydbsmh5j7sg5wn7av9l42drjaq-apcupsd-exporter-0.2.0.drv'...
...
/nix/store/kl2h1zmm4rqry56a3mh5dpbgmgg0v3dk-apcupsd-exporter-0.2.0

[matt@servnerr-3:~/src/nixpkgs]$ ./result/bin/apcupsd_exporter -h
Usage of ./result/bin/apcupsd_exporter:
  -apcupsd.addr string
        address of apcupsd Network Information Server (NIS) (default ":3551")
  -apcupsd.network string
        network of apcupsd Network Information Server (NIS): typically "tcp", "tcp4", or "tcp6" (default "tcp")
  -telemetry.addr string
        address for apcupsd exporter (default ":9162")
  -telemetry.path string
        URL path for surfacing collected metrics (default "/metrics")

[matt@servnerr-3:~/src/nixpkgs]$ ./result/bin/apcupsd_exporter
2020/06/08 10:02:48 starting apcupsd exporter on ":9162" for server tcp://:3551
```

/cc @1000101 and @mmahut per the initial PR: https://github.com/NixOS/nixpkgs/pull/69140

### CODE REVIEW

**Comment by mdlayher** [comment]
@GrahamcOfBorg build prometheus-apcupsd-exporter

File: pkgs/servers/monitoring/prometheus/apcupsd-exporter.nix [hunk 1]
```diff
@@ -1,27 +1,23 @@
...
 
-  doCheck = true;
+  vendorSha256 = "09x8y8pmgfn897hvnk122ry460y12b8a7y5fafri5wn9vxab9r82";
 
   meta = with stdenv.lib; {
     description = "Provides a Prometheus exporter for the apcupsd Network Information Server (NIS)";
     homepage = "https://github.com/mdlayher/apcupsd_exporter";
     license = licenses.mit;
-    maintainers = with maintainers; [ maintainers."1000101" ];
+    maintainers = with maintainers; [ "1000101" mdlayher ];
```
Comment (Mic92) [inline pkgs/servers/monitoring/prometheus/apcupsd-exporter.nix:20]: I think we need to keep it as this: 

```suggestion
    maintainers = with maintainers; [ maintainers."1000101" mdlayher ];
```

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 89809` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- prometheus-apcupsd-exporter
</details>

File: pkgs/servers/monitoring/prometheus/apcupsd-exporter.nix (hunk 1 above)
Comment (mdlayher) [reply pkgs/servers/monitoring/prometheus/apcupsd-exporter.nix:20]: Ack, sorry!

Comment (Mic92) [reply pkgs/servers/monitoring/prometheus/apcupsd-exporter.nix]:
```
nix-repl> builtins.toJSON (with stdenv.lib; with maintainers;  [ maintainers."1000101" mdlayher ])
"[{\"email\":\"b1000101@pm.me\",\"github\":\"1000101\",\"githubId\":791309,\"name\":\"Jan Hrnko\"},{\"email\":\"mdlayher@gmail.com\",\"github\":\"mdlayher\",\"githubId\":1926905,\"name\":\"Matt Layher\"}]"

nix-repl> builtins.toJSON (with stdenv.lib; with maintainers;  [ "1000101" mdlayher ])
"[\"1000101\",{\"email\":\"mdlayher@gmail.com\",\"github\":\"mdlayher\",\"githubId\":1926905,\"name\":\"Matt Layher\"}]"
```

**Pull request merged**